### PR TITLE
Update Azure Arc-enabled vSphere resource bridge scenario

### PR DESCRIPTION
**Pull request by lanicolas**
Refresh of the Azure Arc-enabled vSphere Jumpstart scenario (https://azurearcjumpstart.io/azure_arc_jumpstart/azure_arc_vsphere/resource_bridge/powershell/)

### CODE REVIEW

File: azure_arc_vsphere_jumpstart/resource_bridge/powershell/vCenter_onboarding.ps1 [hunk 1]
```diff
@@ -81,29 +82,101 @@ New-Item -Force -Path "." -Name ".temp" -ItemType "directory" > $null
...
+    $gitUrl = "https://raw.githubusercontent.com/Azure/azure-cli/main/src/azure-cli/setup.py"
+    try {
+        $response = Invoke-WebRequest -Uri $gitUrl -TimeoutSec 30
+    }
+    catch {
+        logWarn "Failed to get the latest version from '$gitUrl': $($_.Exception.Message)"
+        return $null
     }
-    log "64-bit python is already installed"
+    if ($response.StatusCode -ne 200) {
```
Comment (chintalavr) [inline azure_arc_vsphere_jumpstart/resource_bridge/powershell/vCenter_onboarding.ps1:95]: Add line break before this if statement

File: azure_arc_vsphere_jumpstart/resource_bridge/powershell/vCenter_onboarding.ps1 [hunk 2]
```diff
@@ -81,29 +82,101 @@ New-Item -Force -Path "." -Name ".temp" -ItemType "directory" > $null
...
+        logWarn "Failed to fetch the latest version from '$gitUrl' with status code '$($response.StatusCode)' and reason '$($response.StatusDescription)'"
+        return $null
+    }
+    $content = $response.Content
```
Comment (chintalavr) [inline azure_arc_vsphere_jumpstart/resource_bridge/powershell/vCenter_onboarding.ps1:99]: Add a line break before this statement

File: azure_arc_vsphere_jumpstart/resource_bridge/powershell/vCenter_onboarding.ps1 [hunk 3]
```diff
@@ -81,29 +82,101 @@ New-Item -Force -Path "." -Name ".temp" -ItemType "directory" > $null
...
+    foreach ($line in $content -split "`n") {
+        if ($line.StartsWith('VERSION')) {
+            $match = [System.Text.RegularExpressions.Regex]::Match($line, 'VERSION = "(.*)"')
+            if ($match.Success) {
+                return $match.Groups[1].Value
+            }
+        }
+    }
+    logWarn "Failed to extract the latest version from the content of '$gitUrl'"
```
Comment (chintalavr) [inline azure_arc_vsphere_jumpstart/resource_bridge/powershell/vCenter_onboarding.ps1:108]: Add a line break before this statement

File: azure_arc_vsphere_jumpstart/resource_bridge/powershell/vCenter_onboarding.ps1 [hunk 4]
```diff
@@ -81,29 +82,101 @@ New-Item -Force -Path "." -Name ".temp" -ItemType "directory" > $null
...
+        return $true
+    }
+
+    log "$arch-bit Azure CLI is already installed. Checking for updates..."
+    $latestAzVersion = getLatestAzVersion
+    if ($latestAzVersion -and ($latestAzVersion -ne $currentAzVersion)) {
+        log "A newer version of Azure CLI ($latestAzVersion) is available, installing it..."
+        return $true
+    }
+    log "Azure CLI is up to date."
```
Comment (chintalavr) [inline azure_arc_vsphere_jumpstart/resource_bridge/powershell/vCenter_onboarding.ps1:154]: Add a line break before this statement

File: azure_arc_vsphere_jumpstart/resource_bridge/powershell/vCenter_onboarding.ps1 [hunk 5]
```diff
@@ -81,29 +82,101 @@ New-Item -Force -Path "." -Name ".temp" -ItemType "directory" > $null
...
+    Invoke-WebRequest -Uri $azCliMsi -OutFile $azCliMsiPath
+    log "Installing 64 bit Azure CLI"
+    log "This might take a while..."
+    $p = Start-Process msiexec.exe -Wait -Passthru -ArgumentList "/i `"$azCliMsiPath`" /quiet /qn /norestart"
     $exitCode = $p.ExitCode
     if ($exitCode -ne 0) {
-        throw "Python installation failed with exit code $LASTEXITCODE"
+        throw "Azure CLI installation failed with exit code $exitCode. See $msiInstallLogPath for additional details."
     }
+    $azCmdDir = Join-Path $env:ProgramFiles "Microsoft SDKs\Azure\CLI2\wbin"
```
Comment (chintalavr) [inline azure_arc_vsphere_jumpstart/resource_bridge/powershell/vCenter_onboarding.ps1:170]: Add a line break before this statement

File: docs/azure_arc_jumpstart/azure_arc_vsphere/resource_bridge/powershell/_index.md [hunk 6]
```diff
@@ -66,17 +66,17 @@ The following README will guide you on how to use the provided PowerShell script
...
   
-    ![Screenshot of Azure resource provider for vSphere](./01.png)
+    ![Screenshot of Azure resource provider for vSphere](./img/01.png)
 
 ### vCenter Prerequisites
 
-- VMware vSphere vCenter Server v6.7.
+- VMware vSphere vCenter Server v6.7 or 8.
 
 - A virtual network that can provide internet access, directly or through a proxy. It must also be possible for VMs on this network to communicate with the vCenter server on TCP port (usually 443).
```
Comment (chintalavr) [inline docs/azure_arc_jumpstart/azure_arc_vsphere/resource_bridge/powershell/_index.md:75]:
```suggestion
- A virtual network that can provide internet access, directly or through a proxy. It must also be possible for VMs on this network to communicate with the vCenter server on a TCP port (usually 443).
```

File: docs/azure_arc_jumpstart/azure_arc_vsphere/resource_bridge/powershell/_index.md [hunk 7]
```diff
@@ -102,102 +102,100 @@ For you to get familiar with the automation and deployment flow, below is an exp
 
 - Change the environment variables according to your environment:
   - _`location`_ - the Azure region you want to deploy to
-  - _`subscriptionId`_ - your subscription ID
+  - _`subscriptionId`_ - your Azure subscription ID
   - _`resourceGroupName`_ - the name of the Azure resource group you will create your resources in
-  - _`applianceName`_ - a name for the Azure Arc resource bridge appliance
+  - _`applianceName`_ - a name for the Azure Arc resource bridge appliance, this will the name that will be used in Azure.
```
Comment (chintalavr) [inline docs/azure_arc_jumpstart/azure_arc_vsphere/resource_bridge/powershell/_index.md:107]: Consider rephrasing the sentence

File: docs/azure_arc_jumpstart/azure_arc_vsphere/resource_bridge/powershell/_index.md [hunk 8]
```diff
@@ -102,102 +102,100 @@ For you to get familiar with the automation and deployment flow, below is an exp
...
-  - _`dnsServer`_ - DNS server to be used for the appliance
-  - _`gateway`_ - Gateway address to be used for the appliance
-  - _`ipAddressPrefix`_ - Network address in CIDR notation to be used by the appliance
+  - _`datacenter`_ - the vSphere datacenter where the appliance will be deployed. Make sure to include the full path, for example: '/Jumpstart - Datacenter'
+  - _`datastore`_ - the vSphere datastore name where the appliance will be deployed. For example: 'NFS'
+  - _`folder`_ - the vSphere folder where the template and appliance will be deployed. Make sure to include the full path, for example: '/Jumpstart - Datacenter/VMs'
+  - _`dnsServer`_ - IP of the DNS server to be used for the appliance
+  - _`gateway`_ - Gateway IP address to be used for the appliance
+  - _`ipAddressPrefix`_ - Network IP address range in CIDR notation to be used by the appliance
   - _`k8sNodeIpPoolStart`_ - IP range start for the IPs to be used by the appliance
```
Comment (chintalavr) [inline docs/azure_arc_jumpstart/azure_arc_vsphere/resource_bridge/powershell/_index.md:123]:
```suggestion
  - _`k8sNodeIpPoolStart`_ - Starting IP address from the subnet to be used by the appliance for node pool
```

File: docs/azure_arc_jumpstart/azure_arc_vsphere/resource_bridge/powershell/_index.md [hunk 9]
```diff
@@ -102,102 +102,100 @@ For you to get familiar with the automation and deployment flow, below is an exp
...
   - _`k8sNodeIpPoolEnd`_ - IP range end for the IPs to be used by the appliance
```
Comment (chintalavr) [inline docs/azure_arc_jumpstart/azure_arc_vsphere/resource_bridge/powershell/_index.md:124]:
```suggestion
  - _`k8sNodeIpPoolEnd`_ - End IP address from subnet to be used by the appliance for node pool
```

**Review by chintalavr — CHANGES_REQUESTED** [review]
Please review the feedback comments and make changes as needed.

File: docs/azure_arc_jumpstart/azure_arc_vsphere/resource_bridge/powershell/_index.md [hunk 10]
```diff
@@ -66,17 +66,17 @@ The following README will guide you on how to use the provided PowerShell script
   az ad sp show --id 'ac9dc5fe-b644-4832-9d03-d9f1ab70c5f7' --query '{name:displayName,objectId:objectId}'
```
Comment (sebassem) [inline docs/azure_arc_jumpstart/azure_arc_vsphere/resource_bridge/powershell/_index.md:66]:
```suggestion
  az ad sp show --id 'ac9dc5fe-b644-4832-9d03-d9f1ab70c5f7' --query '{name:displayName,objectId:id}'
```

File: docs/azure_arc_jumpstart/azure_arc_vsphere/resource_bridge/powershell/_index.md [hunk 11]
```diff
@@ -66,17 +66,17 @@ The following README will guide you on how to use the provided PowerShell script
...
   ```
   
-    ![Screenshot of Azure resource provider for vSphere](./01.png)
+    ![Screenshot of Azure resource provider for vSphere](./img/01.png)
 
 ### vCenter Prerequisites
 
-- VMware vSphere vCenter Server v6.7.
+- VMware vSphere vCenter Server 8.
```
Comment (sebassem) [inline docs/azure_arc_jumpstart/azure_arc_vsphere/resource_bridge/powershell/_index.md:73]: should we mention 7 or 8 since both are supported ?

**Review by sebassem — APPROVED** [review]
minor comments in readme

File: docs/azure_arc_jumpstart/azure_arc_vsphere/resource_bridge/powershell/_index.md (hunk 11 above)
Comment (lanicolas) [reply docs/azure_arc_jumpstart/azure_arc_vsphere/resource_bridge/powershell/_index.md]: true, fixed

File: docs/azure_arc_jumpstart/azure_arc_vsphere/resource_bridge/powershell/_index.md [hunk 12]
```diff
@@ -102,102 +102,100 @@ For you to get familiar with the automation and deployment flow, below is an exp
...
+  - _`datastore`_ - the vSphere datastore name where the appliance will be deployed. For example: 'NFS'
+  - _`folder`_ - the vSphere folder where the template and appliance will be deployed. Make sure to include the full path, for example: '/Jumpstart - Datacenter/VMs'
+  - _`dnsServer`_ - IP of the DNS server to be used for the appliance
+  - _`gateway`_ - Gateway IP address to be used for the appliance
+  - _`ipAddressPrefix`_ - Network IP address range in CIDR notation to be used by the appliance
+  - _`k8sNodeIpPoolStart`_ - Starting IP address from the subnet to be used by the appliance for node pool
+  - _`k8sNodeIpPoolEnd`_ - End IP address from the subnet to be used by the appliance for node pool
+  - _`segment`_ - Name of the virtual network or segment to which the appliance VM must be connected. Use the same name as it is displayed in your vCenter, for example: 'Jumpstart VM segment - (VLAN 111)'
+  - _`resourcePool`_ - the name of the vSphere resource pool to be used by the appliance. Make sure to include the full path, for example: '/Jumpstart - Datacenter/host/Jumpstart Cluster/Resources'
```
Comment (likamrat) [inline docs/azure_arc_jumpstart/azure_arc_vsphere/resource_bridge/powershell/_index.md:126]:
```suggestion
  - _`datacenter`_ - The vSphere datacenter where the appliance will be deployed. Make sure to include the full path, for example: '/Jumpstart - Datacenter'
  - _`datastore`_ - The vSphere datastore name where the appliance will be deployed. For example: 'NFS'
  - _`folder`_ - the vSphere folder where the template and appliance will be deployed. Make sure to include the full path, for example: '/Jumpstart - Datacenter/VMs'
  - _`dnsServer`_ - IP of the DNS server to be used for the appliance
  - _`gateway`_ - Gateway IP address to be used for the appliance
  - _`ipAddressPrefix`_ - Network IP address range in CIDR notation to be used by the appliance
  - _`k8sNodeIpPoolStart`_ - Starting IP address from the subnet to be used by the appliance for the node pool
  - _`k8sNodeIpPoolEnd`_ - End IP address from the subnet to be used by the appliance for the node pool
  - _`segment`_ - Name of the virtual network or segment to which the appliance VM must be connected. Use the same name as it is displayed in your vCenter, for example: 'Jumpstart VM segment - (VLAN 111)'
  - _`resourcePool`_ - The name of the vSphere resource pool to be used by the appliance. Make sure to include the full path, for example: '/Jumpstart - Datacenter/host/Jumpstart Cluster/Resources'
```

File: docs/azure_arc_jumpstart/azure_arc_vsphere/resource_bridge/powershell/_index.md [hunk 13]
```diff
@@ -102,102 +102,100 @@ For you to get familiar with the automation and deployment flow, below is an exp
 
 - Change the environment variables according to your environment:
   - _`location`_ - the Azure region you want to deploy to
-  - _`subscriptionId`_ - your subscription ID
+  - _`subscriptionId`_ - your Azure subscription ID
   - _`resourceGroupName`_ - the name of the Azure resource group you will create your resources in
-  - _`applianceName`_ - a name for the Azure Arc resource bridge appliance
+  - _`applianceName`_ - a name for the Azure Arc resource bridge appliance, as it will be shown in Azure.
   - _`customLocationName`_ - the name of your Azure Arc custom location
   - _`vcenterName`_ -  the name of your vCenter
```
Comment (likamrat) [inline docs/azure_arc_jumpstart/azure_arc_vsphere/resource_bridge/powershell/_index.md:109]:
```suggestion
  - _`vcenterName`_ -  The name of your vCenter
```
maybe "The fully qualified domain name (FQDN) of your vCenter"?

**Review by likamrat — CHANGES_REQUESTED** [review]
See comments

**Pull request closed without merging**